tests/nested/manual/refresh-revert-gadget-and-kernel/task.yaml

```diff
@@ -0,0 +1,75 @@
+summary: Refresh and revert the gadget and kernel snap
+
+description: |
+    This test validates the kernel and gadget snaps can be refreshed
+    and reverted to the new snaps published to edge channel.
+
+systems: [ubuntu-20.04-*]
+
+environment:
+    CORE_CHANNEL: beta
+    CORE_REFRESH_CHANNEL: edge
+    BUILD_SNAPD_FROM_CURRENT: false
+    USE_CLOUD_INIT: true
+    ENABLE_SECURE_BOOT: true
+    ENABLE_TPM: true
+
+prepare: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    rm -f "$WORK_DIR/image/ubuntu-core.img"
+
+    create_nested_core_vm
+    start_nested_core_vm
+
+restore: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    destroy_nested_vm
+    cleanup_nested_env
+
+    rm -f "$WORK_DIR/image/ubuntu-core.img"
+
+execute: |
+    #shellcheck source=tests/lib/nested.sh
+    . "$TESTSLIB/nested.sh"
+
+    # Check the initial version of the snaps is correct
+    execute_remote "snap list pc" | MATCH "^pc .* 20/$CORE_CHANNEL .*"
+    execute_remote "snap list pc-kernel" | MATCH "^pc-kernel .* 20/$CORE_CHANNEL .*"
+
+    # The kernel and gadget snaps are refreshed
+    execute_remote "sudo snap refresh --$CORE_REFRESH_CHANNEL pc"
+    execute_remote "sudo snap refresh --$CORE_REFRESH_CHANNEL pc-kernel"
+
+    # Check the new version of the snaps is correct
+    execute_remote "snap list pc" | MATCH "^pc .* 20/$CORE_REFRESH_CHANNEL .*"
+    execute_remote "snap list pc-kernel" | MATCH "^pc-kernel .* 20/$CORE_REFRESH_CHANNEL .*"
+
+    # Reboot the system
+    execute_remote "sudo reboot"
```

Comment:
should we rather wait for the system to reboot itself? preferably ask systemd whether the system is going down.

Reply:
Thanks for the review, reboot now is not manually triggered anymore